exec/java-exec/src/main/java/org/apache/drill/exec/service/ServiceEngine.java

```diff
@@ -177,6 +180,16 @@ public void run() {
     });
   }
 
+  /** If we are in distributed mode, disallow using loopback address to register new drillbit
+   *  unless zookeeper is on the same node with drillbit. */
+  private void checkLoopbackAddress(String address) throws DrillbitStartupException, UnknownHostException {
+    String coordAddress = StringUtils.substringBefore(config.getString(ExecConstants.ZK_CONNECTION), ":");
```

Comment:
Your comment, made me realize that I was wrong assuming that in embedded mode zookeeper address will be localhost. Actually it will be the same as set in config. So I have passed flag to ServiceEngine class so we know in which mode drill. Pull request was updated.